cli/commands/site/stop-all.ts

```diff
@@ -0,0 +1,134 @@
+import { connect, disconnect } from 'cli/lib/pm2-manager';
+import { stopProxyIfNoSitesNeedIt } from 'cli/lib/site-utils';
+import { isServerRunning, stopWordPressServer } from 'cli/lib/wordpress-server-manager';
+import { Logger, LoggerError } from 'cli/logger';
+import { StudioArgv } from 'cli/types';
+
+const logger = new Logger< LoggerAction >();
+
+const filterRunningSites = async ( sites: SiteData[] ): Promise< SiteData[] > => {
+	const runningSites = [];
+
+	for ( const site of sites ) {
+		const runningProcess = await isServerRunning( site.id );
+
+		if ( runningProcess ) {
+			runningSites.push( site );
+		}
+	}
+
+	return runningSites;
+};
+
+export async function runCommand(): Promise< void > {
+	try {
+		const appdata = await readAppdata();
+		const allSites = appdata.sites;
+
+		if ( ! allSites.length ) {
+			logger.reportSuccess( __( 'No sites found' ) );
+			return;
+		}
+
+		await connect();
+
+		const runningSites = await filterRunningSites( allSites );
+
+		if ( ! runningSites.length ) {
+			logger.reportSuccess( __( 'No sites are currently running' ) );
+			return;
+		}
+
+		const stoppedSiteIds: string[] = [];
+
+		logger.reportStart(
+			LoggerAction.STOP_ALL_SITES,
+			sprintf(
+				__( 'Stopping all WordPress sites... (%d/%d)' ),
+				stoppedSiteIds.length,
+				runningSites.length
+			)
+		);
+
+		const errors: Array< { siteName: string; error: Error } > = [];
+
+		for ( const site of runningSites ) {
+			try {
+				await stopWordPressServer( site.id );
+				await clearSiteLatestCliPid( site.id );
+
+				stoppedSiteIds.push( site.id );
+
+				logger.reportProgress(
+					sprintf(
+						__( 'Stopping all WordPress sites... (%d/%d)' ),
+						stoppedSiteIds.length,
+						runningSites.length
+					)
+				);
+			} catch ( error ) {
+				errors.push( {
+					siteName: site.name,
+					error: error instanceof Error ? error : new Error( String( error ) ),
+				} );
+			}
+		}
+
+		try {
+			await stopProxyIfNoSitesNeedIt( stoppedSiteIds, logger );
+		} catch ( error ) {
+			// Non-critical error, just log it
+			logger.reportWarning( __( 'Failed to stop proxy server' ) );
+		}
+
+		if ( errors.length === 0 ) {
+			const siteNames = runningSites.map( ( s ) => s.name ).join( ', ' );
+			logger.reportSuccess(
+				sprintf( __( 'Successfully stopped %d site(s): %s' ), runningSites.length, siteNames )
+			);
+		} else if ( errors.length === runningSites.length ) {
+			const failedNames = errors.map( ( e ) => e.siteName ).join( ', ' );
+			throw new LoggerError(
+				sprintf( __( 'Failed to stop all (%d) sites: %s' ), runningSites.length, failedNames )
+			);
+		} else {
+			const successCount = runningSites.length - errors.length;
+			const failedNames = errors.map( ( e ) => e.siteName ).join( ', ' );
+			throw new LoggerError(
+				sprintf(
+					__( 'Stopped %d site(s), but %d failed: %s' ),
+					successCount,
+					errors.length,
+					failedNames
+				)
+			);
+		}
```

Comment:
```suggestion
		if ( stoppedSiteIds.length === runningSites.length ) {
			logger.reportSuccess(
				sprintf(
					_n(
						'Successfully stopped %d site',
						'Successfully stopped %d sites',
						runningSites.length
					),
					runningSites.length
				)
			);
		} else if ( stoppedSiteIds.length === 0 ) {
			throw new LoggerError( __( 'Failed to stop all sites' ) );
		} else {
			throw new LoggerError(
				sprintf( __( 'Stopped %d sites out of %d' ), stoppedSiteIds.length, runningSites.length )
			);
		}
```
If we already log the site names on success or error, we don't need to repeat them here, either. Also, we can use `_n` for singular/plural translations.